modules/authentication/components/RegistrationConfirmed.jsx

```diff
@@ -0,0 +1,61 @@
+import React, {
+  cloneElement,
+  Component,
+  PropTypes
+} from 'react';
+import {
+  Link
+} from 'react-router';
+
+const propTypes = {
+  actions: PropTypes.shape({
+    authentication: PropTypes.shape({
+      updateTokenInfo: PropTypes.func.isRequired
+    }).isRequired
+  }).isRequired,
+  baseClassName: PropTypes.string,
+  children: PropTypes.node,
+  location: PropTypes.shape({
+    query: PropTypes.shape({
+      client_id: PropTypes.string,
+      expiry: PropTypes.string,
+      token: PropTypes.string,
+      uid: PropTypes.string
+    }).isRequired
+  }).isRequired,
+  router: PropTypes.shape({
+    push: PropTypes.func.isRequired
+  })
+};
+
+class RegistrationConfirmed extends Component {
+  componentWillMount() {
+    const { query } = this.props.location;
+    const tokenInfo = {
+      'access-token': query.token,
+      client: query.client_id,
+      expiry: query.expiry,
+      'token-type': 'Bearer',
+      uid: query.uid
+    };
+
+    return Promise.resolve(this.props.actions.authentication.updateTokenInfo(tokenInfo))
```

Comment:
Yes, but I think `updateTokenInfo` should probably be moved up into `services` somewhere and return a Promise that returns an Object rather than an Object, for boilerplate purposes.